test/nodetool/test_status.py

```diff
@@ -0,0 +1,412 @@
+    Joining = 'J'
+    Leaving = 'L'
+    Moving = 'M'
+    Normal = 'N'
+
+
+class Node(NamedTuple):
+    endpoint: str
+    host_id: str
+    load: int
+    tokens: list[str]
+    datacenter: str
+    rack: str
+    status: NodeStatus
+    state: NodeState
+
+
+null_ownership_error = ("Non-system keyspaces don't have the same replication settings, "
+                        "effective ownership information is meaningless")
+
+
+def validate_status_output(res, keyspace, nodes, ownership, resolve):
+    datacenters = sorted(list(set([node.datacenter for node in nodes.values()])))
+    load_multiplier = {"KB": 1024, "MB": 1024**2, "GB": 1024**3, "TB": 1024**4}
+
+    lines = res.split('\n')
+    i = 0
+
+    for dc in datacenters:
+        dc_line = lines[i]
+        assert dc_line.startswith("Datacenter:")
+        res_dc = dc_line.split()[1]
+        assert dc == res_dc
+        dc_line_len = len(dc_line)
+
+        i += 1
+        assert lines[i] == "=" * dc_line_len
+
+        i += 1
+        assert lines[i] == "Status=Up/Down"
+
+        i += 1
+        assert lines[i] == "|/ State=Normal/Leaving/Joining/Moving"
+
+        i += 1
+        if keyspace is None:
+            assert lines[i].split() == ["--", "Address", "Load", "Tokens", "Owns", "Host", "ID", "Rack"]
+        else:
+            assert lines[i].split() == ["--", "Address", "Load", "Tokens", "Owns", "(effective)", "Host", "ID", "Rack"]
+
+        dc_eps = {ep for ep, node in nodes.items() if node.datacenter == dc}
+        if resolve:
+            name_to_ep = {getnameinfo((ep, 0), 0)[0]: ep for ep in nodes.keys()}
+
+        # The legacy nodetool prints endpoints in random order (probably in hash-map order).
+        # So just make sure here, that we see all endpoints from each DC and that their properties are correct.
+        while dc_eps:
+            i += 1
+
+            assert lines[i] != ""
+
+            status_state, ep, load, load_unit, tokens, owns, host_id, rack = lines[i].split()
+
+            if resolve:
+                assert ep in name_to_ep
+                ep = name_to_ep[ep]
+
+            assert ep in dc_eps
+            dc_eps.remove(ep)
+
+            node = nodes[ep]
+
+            assert status_state == "{}{}".format(nodes[ep].status.value, nodes[ep].state.value)
+            assert load == "{:.2f}".format(int(node.load) / load_multiplier[load_unit])
+            assert int(tokens) == len(node.tokens)
+            if keyspace is None:
+                assert owns == "?"
+            else:
+                assert owns == "{:.1f}%".format(float(ownership[ep]) * 100)
+            assert host_id == node.host_id
+            assert rack == node.rack
+
+        assert len(dc_eps) == 0
+        i += 1
+
+    i += 1
+    if keyspace is None:
+        assert lines[i] == f"Note: {null_ownership_error}"
+    else:
+        assert lines[i] == ""
+
+
+def ratio_helper(a: int, b: int):
+    maxint_64 = (2**63)-1
+    minint_64 = -(2**63-1)
+    int64_range = 2**64-1
```

Comment:
Also the range of 64 bits is 2**64 (it's easiest to think about the unsigned case - from 0 to 2**64-1). Unless I'm misunderstanding what you mean by "range".

Reply:
I think I got confused thinking too much about how the sign bit and 2's complement changes things.

Reply:
Fixed in v2.